OpenRA.Mods.Common/Widgets/Logic/Ingame/IngameDebugLogic.cs

```diff
@@ -0,0 +1,54 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Drawing;
+using OpenRA.Network;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets.Logic
+{
+	public class IngameDebugLogic : ChromeLogic
+	{
+		readonly OrderManager orderManager;
+		readonly Ruleset modRules;
+
+		readonly ContainerWidget debugOverlay;
+		readonly ChatDisplayWidget debugOverlayDisplay;
+
+		[ObjectCreator.UseCtor]
+		public IngameDebugLogic(Widget widget, OrderManager orderManager, World world, Ruleset modRules)
+		{
+			this.orderManager = orderManager;
+			this.modRules = modRules;
+
+			var chatPanel = (ContainerWidget)widget;
+			debugOverlay = chatPanel.Get<ContainerWidget>("DEBUG_OVERLAY");
+			debugOverlayDisplay = debugOverlay.Get<ChatDisplayWidget>("DEBUG_DISPLAY");
+
+			foreach (var chatLine in orderManager.ChatCache)
+				AddDebugLine(chatLine.Color, chatLine.Name, chatLine.Text);
```

Comment:
I'm not sure about this. In `IngameChatLogic` when it runs through the cache it *doesn't* add lines or notifications for anything from the cache.